### PR TITLE
Cleanup of the toolshed's url handling methods.

### PR DESCRIPTION
**Pull request by davebx**
Use urllib.urlencode to build the url for various operations.

I've tested this as extensively as I could, but more review would likely not hurt.

### CODE REVIEW

**Comment by bgruening** [comment]
@davebx you should get such a fancy pep8 plugin :)

**Comment by davebx** [comment]
@bgruening You're right. I added a plugin to sublime, now to figure out how it works.

File: lib/galaxy/webapps/galaxy/controllers/admin_toolshed.py [hunk 1]
```diff
@@ -555,12 +555,11 @@ def install_latest_repository_revision( self, trans, **kwd ):
...
-                        params = '?name=%s&owner=%s&changeset_revisions=%s&galaxy_url=%s' % ( name,
-                                                                                              owner,
-                                                                                              str( latest_downloadable_revision ),
-                                                                                              web.url_for( '/', qualified=True ) )
-                        url = common_util.url_join( tool_shed_url,
-                                                    'repository/install_repositories_by_revision%s' % params )
+                        params = dict( name=name,
+                                       owner=owner,
+                                       changeset_revisions=str( latest_downloadable_revision ),
+                                       galaxy_url=web.url_for( '/', qualified=True ) )
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/controllers/admin_toolshed.py:561]: Here you need to redefine `pathspec`:

```
                        pathspec = ['repository', 'install_repositories_by_revision']
```

File: lib/galaxy/webapps/galaxy/controllers/admin_toolshed.py [hunk 2]
```diff
@@ -981,10 +980,10 @@ def prepare_for_install( self, trans, **kwd ):
                 repository = suc.get_tool_shed_repository_by_id( trans.app, updating_repository_id )
                 # For backward compatibility to the 12/20/12 Galaxy release.
                 try:
-                    params = '?name=%s&owner=%s' % ( str( repository.name ), str( repository.owner ) )
-                    url = common_util.url_join( tool_shed_url,
-                                                'repository/get_repository_id%s' % params )
-                    repository_ids = common_util.tool_shed_get( trans.app, tool_shed_url, url )
+                    params = dict( name=str( repository.name ), owner=str( repository.owner ) )
+                    # params = '?name=%s&owner=%s' % ( str( repository.name ), str( repository.owner ) )
```
Comment (nsoranzo) [inline lib/galaxy/webapps/galaxy/controllers/admin_toolshed.py]: This commented line can be removed.

File: lib/tool_shed/galaxy_install/repository_dependencies/repository_dependency_manager.py [hunk 3]
```diff
@@ -378,7 +377,8 @@ def get_required_repo_info_dicts( self, tool_shed_url, repo_info_dicts ):
                     if suc.is_tool_shed_client( self.app ):
                         # Handle secure / insecure Tool Shed URL protocol changes and port changes.
                         tool_shed_url = common_util.get_tool_shed_url_from_tool_shed_registry( self.app, tool_shed_url )
-                    url = common_util.url_join( tool_shed_url, '/repository/get_required_repo_info_dict' )
+                        pathspec = [ 'repository', 'get_required_repo_info_dict' ]
```
Comment (nsoranzo) [inline lib/tool_shed/galaxy_install/repository_dependencies/repository_dependency_manager.py]: This line seems to be over-indented, should be one tab less.

File: lib/tool_shed/util/common_util.py [hunk 4]
```diff
@@ -338,19 +339,22 @@ def tool_shed_get( app, tool_shed_url, uri ):
...
-def url_join( *args ):
+def url_join( base_url, pathspec=None, params=None ):
     """Return a valid URL produced by appending a base URL and a set of request parameters."""
-    parts = []
-    for arg in args:
-        if arg is not None:
-            parts.append( arg.strip( '/' ) )
-    return '/'.join( parts )
+    url = base_url.rstrip( '/' )
+    log.debug( '%s\n%s\n%s', url, pathspec, params )
```
Comment (nsoranzo) [inline lib/tool_shed/util/common_util.py]: Is this logging still needed?

File: lib/tool_shed/util/shed_util_common.py [hunk 5]
```diff
@@ -825,14 +820,10 @@ def get_tool_shed_status_for_installed_repository( app, repository ):
         # using a boolean value.
         log.debug( "Error attempting to get tool shed status for installed repository %s: %s\nAttempting older 'check_for_updates' method.\n" %
                    ( str( repository.name ), str( e ) ) )
-        params = '?name=%s&owner=%s&changeset_revision=%s&from_update_manager=True' % ( str( repository.name ),
-                                                                                        str( repository.owner ),
-                                                                                        str( repository.changeset_revision ) )
-        url = common_util.url_join( tool_shed_url,
-                                    'repository/check_for_updates%s' % params )
```
Comment (nsoranzo) [inline lib/tool_shed/util/shed_util_common.py:832]: Here you are re-using `params` as defined above, but you seem to be missing `from_update_manager=True`.

**Comment by nsoranzo** [comment]
:+1:

**Comment by martenson** [comment]
LGTM. Thanks @davebx for working on this.

**Pull request merged**